controllers/openstackmachine_controller.go

```diff
@@ -364,10 +356,13 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 		conditions.MarkTrue(openStackMachine, infrav1.InstanceReadyCondition)
 		openStackMachine.Status.Ready = true
 	case infrav1.InstanceStateError:
-		// Error is unexpected, thus we report error and never retry
+		// If the machine has a NodeRef the error could be something temporary.
```

Comment:
Can we expand on this comment a bit? The comment is correct, but it might not be immediately obvious why. Could we explain that if the Machine has a NodeRef then we know it was working at some point, and therefore the error (presumably) isn't in the spec.

Reply:
Good point, fixed!